host/src/request.rs

```diff
@@ -1,77 +1,340 @@
+            rpc: value
+                .rpc
+                .ok_or(HostError::InvalidRequestConfig("Missing rpc".to_string()))?,
+            l1_rpc: value.l1_rpc.ok_or(HostError::InvalidRequestConfig(
+                "Missing l1_rpc".to_string(),
+            ))?,
+            beacon_rpc: value.beacon_rpc.ok_or(HostError::InvalidRequestConfig(
+                "Missing beacon_rpc".to_string(),
+            ))?,
+            network: value
+                .network
+                .ok_or(HostError::InvalidRequestConfig(
+                    "Missing network".to_string(),
+                ))?
+                .parse()
+                .map_err(|_| HostError::InvalidRequestConfig("Invalid network".to_string()))?,
+            graffiti: value
+                .graffiti
+                .ok_or(HostError::InvalidRequestConfig(
+                    "Missing graffiti".to_string(),
+                ))?
+                .parse()
+                .map_err(|_| HostError::InvalidRequestConfig("Invalid graffiti".to_string()))?,
+            prover: value
+                .prover
+                .ok_or(HostError::InvalidRequestConfig(
+                    "Missing prover".to_string(),
+                ))?
+                .parse()
+                .map_err(|_| HostError::InvalidRequestConfig("Invalid prover".to_string()))?,
+            proof_type: value
+                .proof_type
+                .ok_or(HostError::InvalidRequestConfig(
+                    "Missing proof_type".to_string(),
+                ))?
+                .parse()
+                .map_err(|_| HostError::InvalidRequestConfig("Invalid proof_type".to_string()))?,
+        })
+    }
 }
 
-#[derive(Debug, Serialize, Deserialize)]
-pub struct JsonRpcRequest<T: Serialize> {
-    pub jsonrpc: String,
-    pub id: serde_json::Value,
-    pub method: String,
-    pub params: T,
+impl ProofRequest {
+    /// Get the instance hash for the protocol instance depending on the proof type.
+    fn instance_hash(&self, pi: ProtocolInstance) -> HostResult<B256> {
+        match self.proof_type {
+            ProofType::Native => Ok(NativeDriver::instance_hash(pi)),
+            ProofType::Sp1 => {
+                #[cfg(feature = "sp1")]
+                return Ok((sp1_prover::Sp1Prover::instance_hash(pi)));
+
+                Err(HostError::FeatureNotSupportedError(self.proof_type.clone()))
+            }
+            ProofType::Risc0 => {
+                #[cfg(feature = "risc0")]
+                return Ok(risc0_prover::Risc0Prover::instance_hash(pi));
+
+                Err(HostError::FeatureNotSupportedError(self.proof_type.clone()))
+            }
+            ProofType::Sgx => {
+                #[cfg(feature = "sgx")]
+                return Ok(sgx_prover::SgxProver::instance_hash(pi));
+
+                Err(HostError::FeatureNotSupportedError(self.proof_type.clone()))
+            }
+        }
+    }
+
+    /// Run the prover driver depending on the proof type.
+    async fn run_driver(
```

Comment:
btw @Brechtpd and I fought on how to name the abstractions and we decided on Prover + Guest. 
In this case Prover == Driver that u mentioned before 😅 so maybe rename this?
```suggestion
    async fn run_prover(
```

Reply:
Should be renamed now